### PR TITLE
feat: deprecate `webrtc` and `quic` features in `libp2p` crate

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

We currently expose `libp2p-quic` and `libp2p-webrtc` as submodules from the `libp2p` crate despite those only being "alpha" status. This causes problems because we need to pin those dependencies due to `cargo` automatically upgrading alphas (which are allowed to incur breaking changes as per semver spec). Additionally, exposing these modules practically hides the "alpha" state of those modules, rendering it kind of obsolete.

The "alpha" state is still true for those modules, thus to properly communicate this to users, we deprecate the modules and require users to spell out the dependency and the alpha version in their manifest.

## Notes & open questions

~This is a breaking change and thus a draft for now. We could also deprecate them first and thus release this instantly. What do you think @mxinden ?~

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: libp2p/Cargo.toml [hunk 1]
```diff
@@ -117,11 +111,8 @@ pin-project = "1.0.0"
 libp2p-deflate = { version = "0.39.0", path = "../transports/deflate", optional = true }
 libp2p-dns = { version = "0.39.0", path = "../transports/dns", optional = true }
 libp2p-mdns = { version = "0.43.0", path = "../protocols/mdns", optional = true }
-libp2p-quic = { version = "=0.7.0-alpha.2", path = "../transports/quic", optional = true }
 libp2p-tcp = { version = "0.39.0", path = "../transports/tcp", optional = true }
-libp2p-tls = { version = "=0.1.0-alpha.2", path = "../transports/tls", optional = true }
```
Comment (mxinden) [inline libp2p/Cargo.toml]: Let's keep `libp2p-tls` now that it is soon no longer an alpha.

https://github.com/libp2p/rust-libp2p/pull/3581/

Comment (thomaseizinger) [reply libp2p/Cargo.toml]: Fixed.

**Comment by thomaseizinger** [comment]
@mxinden I now changed this to deprecate instead of remove the modules which allows us to release this as non-breaking change. Hopefully, this contributes to lessen the problems we have with these alphas.

File: libp2p/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
 # 0.51.1
 
 - Depend on `libp2p-tls` `v0.1.0`.
+- Deprecate the `quic` and `webrtc` feature.
+  These two crates are only in alpha state.
+  To properly communicate this to users, we want them to add the dependency directly which makes the `alpha` version visible.
+  It is recommended to pin the versions of `libp2p-quic` and `libp2p-webrtc` using `=`.
```
Comment (mxinden) [inline libp2p/CHANGELOG.md]:
```suggestion
```

With https://github.com/libp2p/rust-libp2p/pull/3584 pinning should no longer be required, correct?

**Review by mxinden — APPROVED** [review]
I am in favor of this, minus the small comment below.

File: libp2p/CHANGELOG.md (hunk 2 above)
Comment (thomaseizinger) [reply libp2p/CHANGELOG.md]: Ah right, because we will bump the minor versions.

**Comment by thomaseizinger** [comment]
I am removing the dependency on #3615. Based on https://github.com/libp2p/rust-libp2p/pull/3386#discussion_r1134619924, I am assuming that @mxinden is in favor of the new changelog syntax so I am using this here and we can iterate on the exact wording in #3615 after that.

**Pull request merged**